In [1]:
import pandas as pd

df = pd.read_csv('../data/raw/credit_risk_dataset.csv')

df.head(5)

,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
0,22,59000,RENT,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3
1,21,9600,OWN,5.0,EDUCATION,B,1000,11.14,0,0.10,N,2
2,25,9600,MORTGAGE,1.0,MEDICAL,C,5500,12.87,1,0.57,N,3
3,23,65500,RENT,4.0,MEDICAL,C,35000,15.23,1,0.53,N,2
4,24,54400,RENT,8.0,MEDICAL,C,35000,14.27,1,0.55,Y,4


In [2]:
df.info()

df.isna().sum()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32581 entries, 0 to 32580
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   person_age                  32581 non-null  int64  
 1   person_income               32581 non-null  int64  
 2   person_home_ownership       32581 non-null  object 
 3   person_emp_length           31686 non-null  float64
 4   loan_intent                 32581 non-null  object 
 5   loan_grade                  32581 non-null  object 
 6   loan_amnt                   32581 non-null  int64  
 7   loan_int_rate               29465 non-null  float64
 8   loan_status                 32581 non-null  int64  
 9   loan_percent_income         32581 non-null  float64
 10  cb_person_default_on_file   32581 non-null  object 
 11  cb_person_cred_hist_length  32581 non-null  int64  
dtypes: float64(3), int64(5), object(4)
memory usage: 3.0+ MB


person_age                       0
person_income                    0
person_home_ownership            0
person_emp_length              895
loan_intent                      0
loan_grade                       0
loan_amnt                        0
loan_int_rate                 3116
loan_status                      0
loan_percent_income              0
cb_person_default_on_file        0
cb_person_cred_hist_length       0
dtype: int64

In [3]:
df.describe()


,person_age,person_income,person_emp_length,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_cred_hist_length
count,32581.000000,3.258100e+04,31686.000000,32581.000000,29465.000000,32581.000000,32581.000000,32581.000000
mean,27.734600,6.607485e+04,4.789686,9589.371106,11.011695,0.218164,0.170203,5.804211
std,6.348078,6.198312e+04,4.142630,6322.086646,3.240459,0.413006,0.106782,4.055001
min,20.000000,4.000000e+03,0.000000,500.000000,5.420000,0.000000,0.000000,2.000000
25%,23.000000,3.850000e+04,2.000000,5000.000000,7.900000,0.000000,0.090000,3.000000
50%,26.000000,5.500000e+04,4.000000,8000.000000,10.990000,0.000000,0.150000,4.000000
75%,30.000000,7.920000e+04,7.000000,12200.000000,13.470000,0.000000,0.230000,8.000000
max,144.000000,6.000000e+06,123.000000,35000.000000,23.220000,1.000000,0.830000,30.000000


In [4]:
df['person_emp_length'] = df['person_emp_length'].fillna(df['person_emp_length'].median()) 
df['loan_int_rate'] = df['loan_int_rate'].fillna(df['loan_int_rate'].median()) 

df.isnull().sum()

person_age                    0
person_income                 0
person_home_ownership         0
person_emp_length             0
loan_intent                   0
loan_grade                    0
loan_amnt                     0
loan_int_rate                 0
loan_status                   0
loan_percent_income           0
cb_person_default_on_file     0
cb_person_cred_hist_length    0
dtype: int64

In [5]:
df[(df['person_age'] > 100) | (df['person_emp_length'] > 60.0)]

,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
0,22,59000,RENT,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3
81,144,250000,RENT,4.0,VENTURE,C,4800,13.57,0,0.02,N,3
183,144,200000,MORTGAGE,4.0,EDUCATION,B,6000,11.86,0,0.03,N,2
210,21,192000,MORTGAGE,123.0,VENTURE,A,20000,6.54,0,0.10,N,4
575,123,80004,RENT,2.0,EDUCATION,B,20400,10.25,0,0.25,N,3
747,123,78000,RENT,7.0,VENTURE,B,20000,10.99,0,0.26,N,4
32297,144,6000000,MORTGAGE,12.0,PERSONAL,C,5000,12.73,0,0.00,N,25


### 🔍 Diagnóstico de Calidad del Dato y Limpieza

####  Tratamiento de Valores Nulos (Imputación):
* **`loan_int_rate`** ($3.116$ nulos, ~$9.5\%$): Se imputó con la **mediana** ($10.99\%$) por consistencia metodológica, coincidiendo estrechamente con la media ($11.01\%$) dada su distribución simétrica.
* **`person_emp_length`** ($895$ nulos, ~$2.7\%$): Se imputó con la **mediana** ($4.0$ años) para evitar el sesgo generado por registros con valores extremos.

####  Depuración de Errores de Registro (Outliers Críticos):
* **Detección:** Se identificaron $5$ registros con edades biológicamente imposibles (`person_age` de $123$ y $144$ años) y $2$ registros con antigüedad laboral inconsistente (`person_emp_length` de $123$ años en perfiles jóvenes).
* **Acción:** Se eliminaron un total de **$7$ registros** ($0.02\%$ del total de $32.581$), asegurando la integridad del dataset sin comprometer representatividad estadística.



In [11]:
df.describe(include='object')

,person_home_ownership,loan_intent,loan_grade,cb_person_default_on_file
count,32574,32574,32574,32574
unique,4,6,7,2
top,RENT,EDUCATION,A,N
freq,16442,6451,10776,26830


In [12]:
df.to_csv('../data/processed/credit_risk_dataset_cleaned.csv', index=False)

### 🔤 Exploración Categórica y Exportación del Dataset Procesado

#### 1. Consistencia de Variables Categóricas:
* **`loan_intent`**: Presenta $6$ categorías válidas, siendo **`EDUCATION`** la de mayor volumen.
* **`person_home_ownership`**, **`loan_grade`** y **`cb_person_default_on_file`**: No registran valores corruptos ni categorías inconsistentes.

#### 2. Exportación a Datos Procesados:
* El dataset depurado se exporta a la ruta `data/processed/credit_risk_dataset_processed.csv`.
* Este archivo constituye la fuente de datos estandarizada para el entrenamiento de modelos de Machine Learning (`02_modelado_evaluacion.ipynb`) y el consumo analítico en **Power BI**.